### DISCUSSION TITLE
Create INDEX CONCURRENTLY

### DISCUSSION THREAD

**Discussion by jdesouza** [pgpartman/pg_partman#578] [General]
I have created a partitioned table but now I need to change one column type. To do that I will need to create INDEX over the new column and it needs to be in a way it does not lock the table.
Looks like the patitioned table is not allowing creating indexes concurrently.
What's the best strategy to workaround this situation?
Also if I need to create this partition by partition, how to dynamically discover the partitions as CREATE INDEX CONCURRENTLY does not run inside a transaction and all function are transactional?

**Comment by keithf4** [comment]
You could use something like the `show_partitions()` function to dynamically generate some SQL statements to create the index concurrently on each child table. Check out the migration and howto docs for some examples of dynamically generating SQL statements to then run later as individual statement calls

https://github.com/pgpartman/pg_partman/tree/master/doc
